src/devices/cpu/dsp56156/dsp56ops.hxx

```diff
@@ -2855,7 +2856,8 @@ static size_t dsp56156_op_do_2(dsp56156_core* cpustate, const uint16_t op, const
 	/* HACK */
 	if (lValue >= 0xfff0)
 	{
-		cpustate->device->logerror("Dsp56156 : DO_2 operation changed %04x to 0000.\n", lValue);
+		cpustate->device->logerror("Dsp56156 : DO_2 operation changed %04x to 0000, pc:%04x ppc:%04x.\n", lValue, cpustate->PCU.pc, cpustate->ppc);
+		cpustate->device->machine().debug_break();
```

Comment:
Is this break intentional or a left-over?

Reply:
It's intentional but temporary. The whole "hack" section should go away once the CPU core is bug-free enough to not go crazy when processing the geometry in Polygonet Commanders. However, things appear to end up going wrong, resulting in incorrect values being fetched, and a `do` loop being initialized with a loop count of 0xFFFF. There's no loop-break opcode within that particular loop, so the value is pretty clearly wrong.

The debug_break() was added as something like the "canary in the coal mine": Once the game is able to get into the part of attract mode with the large, blinking "Insert Coins" banner without hitting that debug_break call, then I've fixed whatever bug(s) are affecting geometry processing.

Reply:
Yeah, I do remember being quite hairy to mess around with.